torch/_refs/linalg/__init__.py

```diff
@@ -97,6 +97,34 @@ def diagonal(
     return torch.diagonal(input, offset=offset, dim1=dim1, dim2=dim2)
 
 
+def _check_vector_norm_args(
+    x: TensorLikeType, ord: Union[float, int] = 2, dim: Optional[DimsType] = None
+):
+    from torch.fx.experimental.symbolic_shapes import sym_or
+
+    if not (ord < 0.0 or ord == float("inf")):
+        return
+
+    torch._check(
+        sym_or(
+            x.numel() != 0,
+            not isinstance(dim, IntLike) and dim is not None and len(dim) != 0,
+        ),
+        "linalg.vector_norm cannot compute the {ord} norm on an empty tensor "
+        "because the operation does not have an identity",
+    )
+
+    shape = x.shape
+    if dim is not None and not isinstance(dim, IntLike):
+        for d in dim:
+            torch._check(
+                sym_or(x.numel() != 0, d < len(shape) and d >= 0 and shape[d] != 0),
+                "linalg.vector_norm cannot compute the {ord} norm on the "
+                f"dimension {d} because this dimension is empty and the "
+                "operation does not have an identity",
```

Comment:
You don't want to use an f-string outside of a lambda (because it will compute the string even if the check passes)

```suggestion
                lambda: (f"linalg.vector_norm cannot compute the {ord} norm on the "
                f"dimension {d} because this dimension is empty and the "
                "operation does not have an identity"),
```

Reply:
this is intentional torch.compile does not support lambda strings here. 
https://github.com/pytorch/pytorch/issues/149997

Reply:
what we can do , is have a utility that pre check to see if the compile check will be deferred or not, and then calls this or that, or fix the issue above, buts its not related to this change. 
previoulsy it was ok, because we would never defer those runtime asserts. not since we can defer them we hit this limitation.